Document maintainership of MetalLB (#661)
metallb/metallb: website/content/community/maintainers.md

---
title: Maintainers
weight: 1
---

MetalLB was created by Dave Anderson.  We are thankful for his efforts to
create MetalLB as a valuable contribution to the Kubernetes ecosystem.  Ongoing

> MetalLB development and maintenance is run by a team of current maintainers.

Reviewer: ```suggestion
MetalLB development and maintenance is done by a team of maintainers.
```

- "development is run" sounds strange to me.
- I find "current maintainers" confusing. I suggest we simply say "a team of maintainers" (or explain what we mean by "current").

Author: is "led", perhaps?  I imagine developing being done by a larger community, but led (or run, organized, something) by the maintainers team.

I've included your suggested text for now though

Reviewer: "led" SGTM.

Author: changed to "led"

MetalLB maintainers are participants in the project with the ability to approve
changes to the MetalLB code base.  The current list of maintainers can be found
in the [MetalLB OWNERS
file](https://github.com/metallb/metallb/blob/main/OWNERS).

## Maintainer Expectations

An active maintainer should:

* Actively participate in reviewing pull requests and incoming issues.  Note

>   that there are no hard rules on what is “active enough” and this is left up

Reviewer: ```suggestion
  that there are no hard rules on what is "active enough" and this is left up
```

Author: done

  to the judgement of the current group of maintainers.

* Actively participate in discussions about design and the future of the
  project.

> * Take responsibility for backports to appropriate braches for PRs they approve

Reviewer: ```suggestion
* Take responsibility for backports to appropriate branches for PRs they approve
```

Author: done

> * Take responsibility for backports to appropriate braches for PRs they approve

Reviewer: >Take responsibility for backports to appropriate branches for PRs they approve and merge

What do we mean by this? I feel like this may be too specific for a general explanation about the responsibilities of the maintainers.

Author: I mean that if a given maintainer is the one to merge a PR into `main`, it should be up to that maintainer to ensure that change is backported, if appropriate.  Does that make sense?

This assumes there is a defined release branch maintenance policy, clarifying what types of changes should be backported ... i'm not sure if that exists though?

Reviewer: Makes sense to me. I don't think we have such a policy.

>   and merge

Reviewer: ```suggestion
  and merge.
```

Author: done

> * Do their best to follow the follow all code, testing, and design conventions

Reviewer: ```suggestion
* Do their best to follow all code, testing, and design conventions
```

- Remove redundant words.

Author: done

>   as determined by consensus among active maintainers

Reviewer: ```suggestion
  as determined by consensus among active maintainers.
```

Author: done

* Gracefully step down from their maintainership role when they are no longer

>   planning to actively participate in the project

Reviewer: ```suggestion
  planning to actively participate in the project.
```

Author: done

> ## Adding New Maintainers

Reviewer: I'm wondering whether we want to publicly specify a process for adding and removing maintainers. Personally I'm fine with the "democratic" process described here, I'm just pointing out that in my opinion we are committed to such processes if we state them on the website, so we should be OK with that.

Author: It's up to you all if you're happy with this process, but I do like seeing something documented.

Reviewer: I think we're good with your proposed process. @rata @daxmc99 please speak up if you have a problem with committing to this process publicly.

New maintainers are added by consensus among the current group of maintainers.

> This can be done via a private discussion via slack or email.  A majority of

Reviewer: ```suggestion
This can be done via a private discussion via Slack or email.  A majority of
```

Author: done

maintainers should support the addition of the new person, and no single
maintainer should object to adding the new maintainer.

## Removing Maintainers

It is normal for maintainers to come and go based on their other

> responsibilities.  Inactive maintainers may be removed if there is no

Reviewer: Double space before "Inactive.

Author: changed all instances of this.

> expectation of ongoing participation.  It should be made very easy to re-add a

Reviewer: >It should be made very easy to re-add a previous maintainer if they are able to actively participate again.

I agree, though I'm not sure what the purpose of this sentence is in the context of this document.

Author: I was trying to make it clear that removing someone for this reason really should not seem like a big deal.  It's based on some of my past experience where people are very hesitant to let go of a maintainership position, though it shouldn't be that way.  Instead of this sentence, an alternative could be something like:

"If a previously inactive maintainer resumes participation, they should be given quick consideration for re-adding to the team."

or I can just drop it if you still don't like calling this out.

Reviewer: >If a previously inactive maintainer resumes participation, they should be given quick consideration for re-adding to the team.

I think this ^ conveys the message better. I'd maybe change "previously inactive" to "former" though.

Author: done
